modules/irsa/external-deployments-operator.tf

```diff
@@ -1,23 +1,280 @@
+
+data "aws_region" "current" {}
+locals {
+  account_id                                      = var.eks_info.cluster.specs.account_id
+  blobs_s3_bucket_arn                             = "arn:${data.aws_partition.current.partition}:s3:::${local.name_prefix}-blobs"
+  environments_repository                         = "${local.name_prefix}/environment"
+  external_deployments_repository                 = "${local.name_prefix}/${var.external_deployments_operator.repository_suffix}"
+  external_deployments_bucket                     = "${local.name_prefix}-${var.external_deployments_operator.bucket_suffix}"
+  external_deployments_operator_role              = "${local.name_prefix}-${var.external_deployments_operator.role_suffix}"
+  external_deployments_operator_role_needs_policy = var.external_deployments_operator.enabled && (var.external_deployments_operator.grant_in_account_policies || var.external_deployments_operator.grant_assume_any_role)
+  region                                          = coalesce(var.external_deployments_operator.region, data.aws_region.current.name)
+}
+
+data "aws_iam_policy_document" "external_deployments_service_account_assume_role" {
+  count = var.external_deployments_operator.enabled ? 1 : 0
+  statement {
+    sid     = "ServiceAccountAssumeRole"
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+    effect  = "Allow"
+    principals {
+      type        = "Federated"
+      identifiers = [local.oidc_provider_arn]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "${trimprefix(local.oidc_provider_url, "https://")}:aud"
+      values   = ["sts.amazonaws.com"]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "${trimprefix(local.oidc_provider_url, "https://")}:sub"
+      values = [
+        "system:serviceaccount:${var.external_deployments_operator.namespace}:${var.external_deployments_operator.service_account_name}"
+      ]
+    }
+  }
+}
+data "aws_iam_policy_document" "external_deployments_self_sagemaker_assume_role" {
```

Comment:
I don't totally follow, why doesn't the service account just have these permissions granted locally if `grant_in_account_policies`? Why the double hop?

Reply:
This is used in two situations 
1. The operator is built to start with its own role and then assume another role possibly in a different AWS account 
2. When creating limited access short term credentials with a specific policy to hand back to users to allow them to invoke the endpoint or download logs